subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/resolver/ResolverEnvironment.kt

```diff
@@ -101,6 +101,15 @@ val Environment.gradleEnvironmentVariables: Map<String, String>
     get() = stringMap("gradleEnvironmentVariables")
 
 
+/**
+ * Environment flag to enabled short circuiting TAPI resolution when script
+ * is changed outside classpath blocks. Defaults to false.
+ */
+internal
+val Map<String, Any?>?.isShortCircuitEnabled: Boolean
+    get() = (this?.get("gradleKotlinDslScriptDependenciesResolverShortCircuit") as? Boolean) ?: false
```

Comment:
💄 `this?.get("gradleKotlinDslScriptDependenciesResolverShortCircuit") == true` ?